protocols/v2/roles-logic-sv2/src/handlers/job_declaration.rs

```diff
@@ -158,9 +158,7 @@ where
             Ok(JobDeclaration::SubmitSolution(message)) => {
                 info!("Received SubmitSolution");
                 debug!("SubmitSolution: {:?}", message);
-                self_
-                    .safe_lock(|x| x.handle_submit_solution(message))
-                    .map_err(|e| crate::Error::PoisonLock(e.to_string()))?
+                Self::handle_submit_solution(self_, message)
```

Comment:
Why did you treat it differently from all the other handlers?
There's no need to use safe_lock ?

Reply:
Perhaps I don't understand your point. Can you be a bit more exahaustive? 
Here it must submit the block to the rpc client. We want that to be async, that would make the trait async. I opted to relaying the message to the caller JobDeclaratorDownstream::start, that is managed by a task in main.  This solution looked easier to me.